### ISSUE TITLE
what is the learning rate when start self-critical training?

### ISSUE THREAD

**Issue by siyilingting** [ruotianluo/self-critical.pytorch#26]
Hi, ruotian. Thank you for your job. I am a bit confused about the learning rate (5e-5) for self-critical training. 
First, I train without self-critical, set `--learning_rate 5e-4 --learning_rate_decay_start 0 --scheduled_sampling_start 0 --max_epochs 30` as you write. Then after 30 epochs I train using self-critical with the setting `--learning_rate 5e-5 --start_from log_fc_rl`, no `learning_rate_decay_start `and `scheduled_sampling_start`. 
In the train.py, I find that https://github.com/ruotianluo/self-critical.pytorch/blob/275e22ccbcf847e77c00b1510b01ab1887438042/train.py#L82
`if vars(opt).get('start_from', None) is not None and os.path.isfile(os.path.join(opt.start_from,"optimizer.pth")):
        optimizer.load_state_dict(torch.load(os.path.join(opt.start_from, 'optimizer.pth')))`.
So does this mean that the learning_rate 5e-5 are discarded, and it's unnecessary for us to set this parameter when we train using self-critical?

**Comment by ruotianluo** [comment]
The learning rate will be updated according to the argument.

**Comment by siyilingting** [comment]
Thank you. But I am still puzzled which learning rate are used in self-critical, the 5e-5 or the one stored in optimizer.pth.

**Comment by ruotianluo** [comment]
I take it back. After reviewing my code, you are correct. This is definitely a bug. The learning rate will be what is saved in the optimizer.

**Comment by ruotianluo** [comment]
I fix it in the latest commit. Thank you for pointing this out.
